shortcutgen.go

```diff
@@ -567,20 +567,28 @@ func makeAggrandizement(valueType *tokenType, variable *varValue, getAs string)
 }
 
 // mapInlineVars finds occurrences of ObjectReplaceChar and adds them to inlineVars to map the inline variables in noVarString.
+// Uses UTF-16 code unit positions to match Apple Shortcuts' encoding requirements.
 func mapInlineVars(noVarString *string) {
 	var variableIdx int
-	var i = 0
-	for _, c := range *noVarString {
-		if c == ObjectReplaceChar {
+	var utf16Pos = 0
+
+	for _, r := range *noVarString {
+		if r == ObjectReplaceChar {
 			inlineVars = append(inlineVars, inlineVar{
 				identifier: varIndex[variableIdx].identifier,
-				col:        i,
+				col:        utf16Pos,
```

Comment:
We should rename this to `charPos` or revert to `i`, because although UTF-16 is relevant, what we are doing is adding 1 or 2 to the character position in the string, not setting the UTF-16 character position in the code point.

```suggestion
	var charPos = 0

	for _, r := range *noVarString {
		if r == ObjectReplaceChar {
inlineVars = append(inlineVars, inlineVar{
identifier: varIndex[variableIdx].identifier,
				col:        charPos,
```